instrumentation/google.golang.org/grpc/otelgrpc/test/interceptor_test.go

```diff
@@ -340,408 +334,7 @@ func eventAttrMap(events []trace.Event) []map[attribute.Key]attribute.Value {
-	name := "github.com.serviceName/bar"
-
-	// create a context with cancel
-	cancelCtx, cancel := context.WithCancel(context.Background())
-	defer cancel()
-	streamClient, err := streamCI(
-		cancelCtx,
-		&grpc.StreamDesc{ServerStreams: true},
-		clientConn,
-		method,
-		func(ctx context.Context,
-			desc *grpc.StreamDesc,
-			cc *grpc.ClientConn,
-			method string,
-			opts ...grpc.CallOption,
-		) (grpc.ClientStream, error) {
-			mockClStr = &mockClientStream{Desc: desc, Ctx: ctx}
-			return mockClStr, nil
-		},
-	)
-	require.NoError(t, err, "initialize grpc stream client")
-	_, ok := getSpanFromRecorder(sr, name)
-	require.False(t, ok, "span should not ended while stream is open")
-
-	req := &grpc_testing.SimpleRequest{}
-	reply := &grpc_testing.SimpleResponse{}
-
-	// send and receive fake data
-	for i := 0; i < 10; i++ {
-		_ = streamClient.SendMsg(req)
-		_ = streamClient.RecvMsg(reply)
-	}
-
-	// close client stream
-	_ = streamClient.CloseSend()
-}
-
-// TestStreamClientInterceptorWithError tests a situation that streamer returns an error.
-func TestStreamClientInterceptorWithError(t *testing.T) {
-	defer goleak.VerifyNone(t)
-
-	clientConn, err := grpc.NewClient("fake:8906",
-		grpc.WithContextDialer(ctxDialer()),
-		grpc.WithTransportCredentials(insecure.NewCredentials()),
-	)
-	if err != nil {
-		t.Fatalf("failed to create client connection: %v", err)
-	}
-	defer clientConn.Close()
-
-	// tracer
-	sr := tracetest.NewSpanRecorder()
-	tp := trace.NewTracerProvider(trace.WithSpanProcessor(sr))
-	//nolint:staticcheck // Interceptors are deprecated and will be removed in the next release.
-	streamCI := otelgrpc.StreamClientInterceptor(
-		otelgrpc.WithTracerProvider(tp),
-		otelgrpc.WithMessageEvents(otelgrpc.ReceivedEvents, otelgrpc.SentEvents),
-	)
-
-	var mockClStr *mockClientStream
-	method := "/github.com.serviceName/bar"
-	name := "github.com.serviceName/bar"
-
-	streamClient, err := streamCI(
-		context.Background(),
-		&grpc.StreamDesc{ServerStreams: true},
-		clientConn,
-		method,
-		func(ctx context.Context,
-			desc *grpc.StreamDesc,
-			cc *grpc.ClientConn,
-			method string,
-			opts ...grpc.CallOption,
-		) (grpc.ClientStream, error) {
-			mockClStr = &mockClientStream{Desc: desc, Ctx: ctx}
-			return mockClStr, errors.New("test")
-		},
-	)
-	require.Error(t, err, "initialize grpc stream client")
-	assert.IsType(t, &mockClientStream{}, streamClient)
-
-	span, ok := getSpanFromRecorder(sr, name)
-	require.True(t, ok, "missing span %s", name)
-
-	expectedAttr := []attribute.KeyValue{
-		semconv.RPCSystemGRPC,
-		otelgrpc.GRPCStatusCodeKey.Int64(int64(grpc_codes.Unknown)),
-		semconv.RPCService("github.com.serviceName"),
-		semconv.RPCMethod("bar"),
-		semconv.NetPeerName("fake"),
-		semconv.NetPeerPort(8906),
-	}
-	assert.ElementsMatch(t, expectedAttr, span.Attributes())
-	assert.Equal(t, codes.Error, span.Status().Code)
-}
-
+// serverChecks can't be removed in #7105. Should be removed when issue #7106 is fixed.
```

Comment:
```suggestion
// TODO: Remove when issue #7106 is fixed.
```

Reply:
#7106 has been fixed, since it's closed. Should `serverChecks` be removed?